clang/include/clang/Basic/LangOptions.h

```diff
@@ -190,10 +190,22 @@ class LangOptions : public LangOptionsBase {
                   /// (More C compatible than offset)
   };
 
+  enum CheriCapIntConversion {
+    CapInt_Explicit, /// Require explicit casts for pointer -> capability and
+                     /// integer -> capability pointer
+    CapInt_Strict,  /// All conversions (including (u)intcap<->int need explicit
+                    /// casts)
+    CapInt_Address, /// Conversions return the address or create an untagged
+                    /// capability with the address
+    CapInt_Relative, /// Perform conversions relative to DDC (previous
+                     /// default)
+    CapInt_Invalid = -1,
+  };
+
   // Work around for bugs in the expression classification which can cause
```

Comment:
Do we have test cases for these bugs somewhere?

Reply:
Not in this repository. I'll inquire whether this is still needed.

Reply:
We can remove this. If we run into issues, we can fix getRealReferenceType() instead of using this workaround.